core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -612,21 +612,42 @@ public ClientFactoryBuilder http1MaxChunkSize(int http1MaxChunkSize) {
 
     /**
      * Sets the idle timeout of a socket connection. The connection is closed if there is no request in
-     * progress for this amount of time.
+     * progress for the given amount of time. By default, HTTP/2 PING frames do not prevent connection from
+     * closing. Use the method {@link ClientFactoryBuilder#idleTimeout(Duration, boolean)} to set whether to
+     * prevent connection from closing when an HTTP/2 PING frame is received.
      */
     public ClientFactoryBuilder idleTimeout(Duration idleTimeout) {
-        requireNonNull(idleTimeout, "idleTimeout");
-        checkArgument(!idleTimeout.isNegative(), "idleTimeout: %s (expected: >= 0)", idleTimeout);
-        return idleTimeoutMillis(idleTimeout.toMillis());
+        return idleTimeout(idleTimeout, false);
+    }
+
+    /**
+     * Sets the idle timeout of a socket connection and whether to reset idle timeout when an HTTP/2 PING
+     * frame or OPTIONS * is received. If `keepAliveOnPing` is `true`, idle timeout is reset
+     * when an HTTP/2 PING frame or OPTIONS * is received.
+     */
+    @UnstableApi
+    public ClientFactoryBuilder idleTimeout(Duration idleTimeout, boolean keepAliveOnPing) {
+        return idleTimeoutMillis(requireNonNull(idleTimeout, "idleTimeout").toMillis(), keepAliveOnPing);
     }
 
     /**
      * Sets the idle timeout of a socket connection in milliseconds. The connection is closed if there is no
-     * request in progress for this amount of time.
+     * request in progress for the given amount of time.
      */
     public ClientFactoryBuilder idleTimeoutMillis(long idleTimeoutMillis) {
+        return idleTimeoutMillis(idleTimeoutMillis, false);
```

Comment:
Should we respect the default value set via `FlagsProvider`?
```suggestion
        return idleTimeoutMillis(idleTimeoutMillis, Flags.defaultClientKeepAliveOnPing());
```
or directly set `ClientFactoryOptions.IDLE_TIMEOUT_MILLIS` instead of delegation.

Reply:
Oops, good idea 👍